core/org.wso2.carbon.tomcat.ext/src/main/java/org/wso2/carbon/tomcat/ext/valves/CompositeValve.java

```diff
@@ -64,7 +64,7 @@ public void invoke(Request request, Response response) throws IOException, Servl
             // --------- Valve chaining happens from here onwards --------------------
 
         } catch (NullPointerException e) {
-            log.error("Could not handle the request, could be due to the maxHttpHeaderSize limitation. ", e);
+            log.error("Could not handle the request, could be due to the maxHttpHeaderSize limitation.");
```

Comment:
Since this is a generic exception, can we special check the maxHttpHeaderSize error and skip the stacktrace. Also we have to keep the stacktrace in debug mode.

Reply:
Also may be we can add the name of the cookie as well. Also better to have the stacktrace as a debug log. WDYT ?

Reply:
do we have enough information to know the header name ?

Reply:
Have added the the `stacktrace` to log in he debug mode.